legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java

```diff
@@ -113,7 +117,9 @@ public interface FunctionExecutionSupport
     String GENERATE_EXECUTION_PLAN_ID = "legend.executionPlan.generate";
     String GRAMMAR_TO_JSON_LAMBDA_ID = "legend.grammarToJson.lambda";
     String JSON_TO_GRAMMAR_LAMBDA_BATCH_ID = "legend.jsonToGrammar.lambda.batch";
+    String GRAMMAR_TO_JSON_VALUE_SPECIFICATION_BATCH_ID = "legend.grammarToJson.valueSpecification.batch";
```

Comment:
Why we need this for?

Reply:
The engine API has separate endpoints for converting grammar to json for value specs, lambdas, and models. If we want, we could just have 1 endpoint on the LSP and perhaps pass an argument indicating the type of the object we are trying to convert?

Reply:
Updated the code as discussed. If the client needs to convert a value spec grammar to json, it will prepend a `|` character to convert it to a lambda, then call the convert grammar to lambda batch function and extract the first element of the body to get the value spec back.